airgun/widgets.py

```diff
@@ -2180,8 +2180,10 @@ def fill(self, item):
         if self.header.is_displayed:
             if self.drop_down.is_displayed:
                 self.drop_down.fill(item)
-            else:
+            elif self.input_box.is_displayed:
                 self.input_box.fill(item)
+            else:
+                self.textarea.fill(item)
```

Comment:
Hmm... Wonder if we shouldn't refactor it into something like:
```python
for widget_name in ('input_box', 'textarea', 'drop_down'):
    widget = getattr(self, widget_name)
    if widget.is_displayed:
        widget.fill(item)
        break
else:
    raise ReadOnlyWidgetError('Setting FooBar is read-only')
```
This should make it easier to add new widget types in the future, in case they appear.

It also correctly handles settings like Authentication -> OAuth consumer key, which are read-only, but which we have treated as simple inputs in the past.

Reply:
I'll try your suggestion as it's more clean and update accordingly if it works well.

Reply:
I think I was a little too fast with that "correctly handles read-only settings". Exception should be raised inside else to `if self.header.is_displayed:`, not to `for widget_name in ():`.

Reply:
@mirekdlugosz, tried your implementation and it works fine.  Hopefully the code is up to standard.

Reply:
Yes, it's very good. Thank you.